.github/workflows/config.yml

```diff
@@ -90,35 +77,51 @@ jobs:
           steps.cache-msys2.outputs.cache-hit != 'true'
         shell: python scripts\msys-bash.py {0}
         run: |
-           scripts/list-build-dependencies.sh -m msys2 -c clang ${{ matrix.conf.without_packages }} | xargs pacman -S --noconfirm
+           pacman -S --noconfirm \
+             meson \
+             mingw-w64-x86_64-ccache \
```

Comment:
Would be nice to have these in a text file too, if there's an equivalent `$(cat list)` for powershell or cmd.exe. Also handy for users trying mingw builds outside the CI environment.

Reply:
I tried that and had a problem - whenever I tried to use a file and pass it to pacman either using `$(cat list)` or `xargs` - it resulted in error. I will experiment more with it. `dnf` or `apt` do not have such problem - it might be a bug in `pacman` (I suspect it works when package names are separated by spaces and fails when they are separated by newlines).

Reply:
I recall having trouble with this too back when setting up our early msys2 CI run and later config heavy. 

>  it works when package names are separated by spaces and fails when they are separated by newlines

This is exactly what xargs will do, and suspect Windows shell is fouling up the pipe. I recall my breakthrough was switching to a bash shell.

Here's a quick "proof", if any such thing exists :-)

Arguments with spaces, and octal-dump of raw byte values:

``` shell
echo 1 2 3
123

echo 1 2 3 | od
0000000 020061 020062 005063
0000006
```

Add xargs:
``` shell
echo 1 2 3 | xargs echo
1 2 3

echo 1 2 3 | xargs echo | od
0000000 020061 020062 005063 <~~ looks good
0000006
```

Now a newline file:
``` shell
cat three_lines.txt
1
2
3

 cat three_lines.txt | od
0000000 005061 005062 005063 <~~ not right
0000006

cat three_lines.txt | xargs echo | od           0000000 020061 020062 005063 <~~ correct
0000006

(Or using < to push-pipe in)

< three_lines.txt xargs echo | od
0000000 020061 020062 005063 <~~ still good
0000006
```

So hopefully you can launch that command in a bash shell. I remember having to use your handy python wrapper shell before these simple commands worked sanely.

Reply:
For the record, `pacman -S - <packages.txt` should work. Since pacman 5.1 / https://bugs.archlinux.org/task/52992 stdin is parsed as newline-separated ONLY, before then it was "whitespace-separated" which sometimes caused problems due to input that could contain embedded whitespace but not embedded newlines.

Reply:
@eli-schwartz Thank you! Yes, this solution worked.

For `dnf`, `apt`, and `brew` we can do `<tool> install $(cat packages.txt)`, where packages.txt is newline-separated - but for `pacman` in MSYS2 it does not work (I don't know about `pacman` in Arch, but I would be surprised if it did not work in Arch either). Maybe it's a fault of pacman, or maybe a fault of MSYS2 - hard to say - I tried several solution and whenever bash subshell is created, it does not work, and using xargs does not work either. It worked with out older script authored by @kcgen because the script was generating space-separated output.

Solutions that didn't work (in all cases packages.txt was newline-separated):

- `pacman -S --noconfirm $(cat packages.txt)`
- `pacman -S --noconfirm $(cat packages.txt | echo)`
- `pacman -S --noconfirm $(cat packages.txt | xargs)`
- `pacman -S --noconfirm $(cat packages.txt | tr '\n' ' ')`
- `cat packages.txt | xargs pacman -S --noconfirm`
- `xargs -a packages.txt pacman -S --noconfirm`

In the end these two solutions work:

1. Having packages.txt space-separated and running `pacman -S --noconfirm $(cat packages.txt)`
2. Having packages.txt newline-separated and running `pacman -S --noconfirm - < packages.txt`

I will stick to solution (2).

Reply:
@eli-schwartz For a moment I thought your solution worked, but it does not…

For `dnf`, `apt`, and `brew` we can do `<tool> install $(cat packages.txt)`, where packages.txt is newline-separated - but for `pacman` in MSYS2 it does not work (I don't know about `pacman` in Arch, but I would be surprised if it did not work in Arch either). Maybe it's a fault of pacman, or maybe a fault of MSYS2 - hard to say - I tried several solution and whenever bash subshell is created, it does not work, and using xargs does not work either. It worked with out older script authored by @kcgen because the script was generating space-separated output.

Solutions that didn't work (in all cases packages.txt was newline-separated):

- `pacman -S --noconfirm $(cat packages.txt)`
- `pacman -S --noconfirm $(cat packages.txt | echo)`
- `pacman -S --noconfirm $(cat packages.txt | xargs)`
- `pacman -S --noconfirm $(cat packages.txt | tr '\n' ' ')`
- `cat packages.txt | xargs pacman -S --noconfirm`
- `xargs -a packages.txt pacman -S --noconfirm`
- `pacman -S --noconfirm - < packages.txt`

This solution works:

- Having packages.txt space-separated and running `pacman -S --noconfirm $(cat packages.txt)`

But well, having package list space separated is annoying… I will probably create a temporary file :/

Reply:
![fuckthis-angry](https://user-images.githubusercontent.com/1557255/105641154-6ffd1000-5e37-11eb-957d-3b71660a71a4.gif)

Reply:
Mystery solved - the problem was caused by Windows newlines. I strategically placed .gitattributes file to prevent Git from converting packages.txt file from Unix line endings to Windows line endings. Fixed by 0ae100ce7a2ddba02bc9c867841031990c41a02f